configure.ac

```diff
@@ -851,6 +851,21 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <malloc.h>]],
  [ AC_MSG_RESULT(no)]
 )
 
+dnl Check for posix_fallocate
+AC_MSG_CHECKING(for posix_fallocate)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+                   #ifdef __linux__
```

Comment:
Not sure this is correct. Is POSIX_C_SOURCE a Linux or a general GCC thing?

Reply:
`_POSIX_C_SOURCE` is not Linux or GCC specific. However, only the [Linux man page](https://linux.die.net/man/3/posix_fallocate) for `posix_fallocate()` mentions it as a requirement. [FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=posix_fallocate) and [NetBSD](https://netbsd.gw.com/cgi-bin/man-cgi?posix_fallocate++NetBSD-current) man pages don't, implying that the function is available without having to define `_POSIX_C_SOURCE` to a specific value.

In any case, this is what the source code in [src/util/system.cpp already does](https://github.com/bitcoin/bitcoin/blob/master/src/util/system.cpp#L21-L29) and it's important to have the check and source code do the same thing.

Reply:
Maybe worth adding comments cross-referencing the identical setups?

Reply:
[Done](https://github.com/bitcoin/bitcoin/compare/14f11083e4ed733520872fb6da3fe59182745db9..182dbdf0f4b6e6484b0d4588aaefacc75862a99c)